File: java/deepspace/src/deepspace/ShieldBooster.java
package deepspace;

/**
 * @brief Representa a los potenciadores de escudo que pueden tener las estaciones espaciales.
 */
class ShieldBooster {
    
    private String name;
    private float boost;
    private int uses;

    //__constructores
    ShieldBooster(String name, float boost, int uses) {
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster otro) {
        name = otro.name;
        boost = otro.boost;
        uses = otro.uses;
    }
    //__ fin constructores 
    
    public float getBoost() { return boost; }
    public int getUses() { return uses; }
    
    public float useIt() {
        if(uses > 0f) {
            uses--;
            return boost;
        }
        return 1.0f;
    }

    /**
       @brief Representación de la clase
       @return Devuelve en un string la representación de la clase 
     */
    public String toString(){

    String representation = "name: " + name + "\n"
	+ "uses: " + uses + "\n"
	+ "boost: " + boost + "\n" ;
    
	return representation; 
    }
    
    ShieldToUI getUIversion(){ return new ShieldToUI(this); }
}

File: ruby/ShieldBooster.rb
require_relative 'ShieldToUI'

module Deepspace

  # @brief representa a los potenciadores de escudo que pueden tener las estaciones espaciales
  class ShieldBooster

    def initialize(name,boost,uses)
      @name = name
      @boost = boost
      @uses = uses
    end

    attr_reader :name, :boost, :uses


    def self.newCopy(s)
      new(s.name,s.boost,s.uses)
    end

    def useIt
      if(uses>0)
        @uses = uses-1
        return boost
      else
        return 1.0
      end
    end

    def getUIversion
      ShieldToUI.new(self)
    end

    def getBoost
      @boost
    end

    def getUses
      @uses
    end

    def to_s
      "Atributos de la instancia de ShieldBooster: \n" +
        "  boost: #{boost}\n" +
        "  uses: #{uses}\n"
    end

    def ShieldToUIgetVersion
      Deepspace::ShieldToUI.new self
    end

  end #class ShieldBooster

end #module Deepspace
